### ISSUE TITLE
About the self.refine_J  and the self.ref_real_J

### ISSUE THREAD

**Issue by hello-trouble** [xiaofeng94/RefineDNet-for-dehazing#4]
Thank you for your excellent job about the single image dehazing. I am a little puzzled about the self.refine_J  and the self.ref_real_J.  For self.refine_J , the input of netRefiner_J is the paired images  (self.real_I, self.dcp_J ) with similar content,  However, For the self.ref_real_J, The input images pair (self.real_I, self.real_J) seems with different content. I am a little puzzled about this issue , Does it affect the performance ? Why not use the image pair (self.real_I, self.refine_J) as the input of the netRefiner_J or drop the self.real_I in the two refined operations. then construct the idt loss.
 Looking forward to your help and thank you very much in advance.

**Comment by xiaofeng94** [comment]
Hey,

Sorry for being late.
Not quite sure I understand the question. But the input of netRefiner_J is the pair of self.real_I and self.dcp_J.
The input of self.netD are unpared self.real_J and fake_J. fake_J comes from a pool where the previous fake_J is stored and randomly sampled.

Hope this helps.

**Comment by hello-trouble** [comment]
Thank you for your kindly reply. I am puzzled about the idendity loss.
# Identity loss, ||refiner_J(real_J) - real_J||
  self.ref_real_J = self.netRefiner_J(torch.cat((self.real_I, self.real_J), 1))
It seems that self.real_I and self.real_J are unpaired. 

> Hey,
> 
> Sorry for being late. Not quite sure I understand the question. But the input of netRefiner_J is the pair of self.real_I and self.dcp_J. The input of self.netD are unpared self.real_J and fake_J. fake_J comes from a pool where the previous fake_J is stored and randomly sampled.
> 
> Hope this helps.



> Hey,
> 
> Sorry for being late. Not quite sure I understand the question. But the input of netRefiner_J is the pair of self.real_I and self.dcp_J. The input of self.netD are unpared self.real_J and fake_J. fake_J comes from a pool where the previous fake_J is stored and randomly sampled.
> 
> Hope this helps.

Thank you for your kindly reply. I am puzzled about the idendity loss.
# Identity loss, ||refiner_J(real_J) - real_J||
  self.ref_real_J = self.netRefiner_J(torch.cat((self.real_I, self.real_J), 1))
It seems that self.real_I and self.real_J are unpaired.

**Comment by hello-trouble** [comment]
您好，非常感谢您的讲解。我对the idendity loss这里有点疑惑，在backward_G的122行，  self.ref_real_J = self.netRefiner_J(torch.cat((self.real_I, self.real_J), 1))
这里self.real_J 和self.real_I是内容相同的清晰图像和雾图吗？

**Comment by xiaofeng94** [comment]
> 您好，非常感谢您的讲解。我对the idendity loss这里有点疑惑，在backward_G的122行， self.ref_real_J = self.netRefiner_J(torch.cat((self.real_I, self.real_J), 1)) 这里self.real_J 和self.real_I是内容相同的清晰图像和雾图吗？

No, paired images are not required.

**Comment by hello-trouble** [comment]
谢谢您，感觉这样用挺奇怪的。在forward() 函数中，netRefiner_J输入是相同内容的图像，self.refine_J = self.netRefiner_J(torch.cat((self.real_I, self.dcp_J), 1))。而在backward_G（）中， self.ref_real_J = self.netRefiner_J(torch.cat((self.real_I, self.real_J), 1)) 输入的是两张内容没有任何联系的图像，感觉这样用挺不好理解的，那在self.netRefiner_J中输入self.real_I是出于什么考虑呢？

**Comment by xiaofeng94** [comment]
Hey,

The input of self.netRefiner_J is designed as a combination of a clear image and a degraded image. We do not acquire the clear image and the degraded image share the same content, but they can be the same. You may have your own design as well.

**Comment by hello-trouble** [comment]
谢谢您的指点。
